Api/Controllers/EventsController.cs

```diff
@@ -0,0 +1,150 @@
+﻿using Api.DTO;
+using Api.DTO.Events;
+using Api.DTO.Users;
+using Domain.Abstractions;
+using Domain.DataModel;
+using Domain.Services;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+
+namespace Api.Controllers;
+
+[Route("events")]
+[ApiController]
+public class EventsController : ControllerBase
+{
+    private IEventService eventService;
+
+    public EventsController(IEventService eventService)
+    {
+        this.eventService = eventService;
+    }
+
+
+    [HttpGet]
+    [Produces("application/json")]
+    public ActionResult<Paged<ListEventDto>> GetEvents([FromQuery] Paging paging, string evNameFilter = "", string orgNameFilter = "", 
+        PriceFilter priceFilter = PriceFilter.Any, int minCapacityFilter = 0, int maxCapacityFilter = int.MaxValue, StartTimeFilter startTimeFilter = StartTimeFilter.Any, bool onlyAvailablePlace = false)
+    {
+        var events = eventService.GetAll();
+        events = Filter(events, evNameFilter, orgNameFilter, priceFilter, minCapacityFilter, maxCapacityFilter, startTimeFilter, onlyAvailablePlace);
+
+        var paged = Paged<ListEventDto>.PageFrom(events.Select(e => e.ToListEventDto()),
+            EventStateComparer.Instance, paging);
+
+        return Ok(paged);
+    }
+
+    [HttpGet]
+    [Route("details")]
+    [Produces("application/json")]
+    public ActionResult GetEvent(Guid guid)
+    {
+        Event @event;
+        try
+        {
+            @event = eventService.GetEvent(guid);
+        }
+        catch (InvalidGuidException ex) 
+        {
+            return BadRequest(ex.Message);
+        }
+
+        return Ok(@event.ToDto());
+    }
+
+    [HttpPost]
+    [Authorize]
+    [Route("create")]
+    public ActionResult CreateEvent(CreateEvent newEvent)
+    {
+        EventCategory cat;
+        if (!Enum.TryParse(newEvent.EventCategory, out cat))
+            return BadRequest("Nonexistent category");
+        try
+        {
+            eventService.AsUser(User.GetGuid()).CreateEvent(newEvent.Title, newEvent.Description, cat, newEvent.PublicationDate, newEvent.StartDate, newEvent.EndDate, newEvent.Location, newEvent.Capacity, newEvent.Fee);
+        }
+        catch (UserUnauthorizedException ex)
+        {
+            return BadRequest(ex.Message);
+        }
+        return Ok();
+    }
+
+    [HttpDelete]
+    [Authorize]
+    [Route("delete")]
+    public ActionResult DeleteEvent(Guid eventGuid)
+    {
+        var e = eventService.AsUser(User.GetGuid()).GetEvent(eventGuid);
+        if (e is null)
+            return BadRequest("Invalid event guid");
+        if (e.OrganizerId != User.GetGuid())
+            return BadRequest("Unauthorized access");
+
+        eventService.AsUser(User.GetGuid()).DeleteEvent(eventGuid);
+        return Ok();
+    }
```

Comment:
I think it is enough to just call DeleteEvent, checking if event guid is wrong or if user is unauthorized is checked inside it already (by GetOrThrow method and AllowOnlyUser method).